statsmodels/regression/tests/test_heckman.py

```diff
@@ -0,0 +1,636 @@
+              'AX' :  .0634079,
+             'AX2' :  .0018978,
+              'WE' :  .1008091,
+             'CIT' :  .3207769,
+           'const' :  2.087786
+           }
+
+    stata_select_coef = {
+              'WA' :   .1903152,
+             'WA2' :  -.0024649,
+          'FAMINC' :   4.22e-06,
+              'WE' :   .0989903,
+               'K' :  -.4578685,
+           'const' :  -4.306956
+           }
+
+    stata_select_stderr = {
+              'WA' :  .0661414,
+             'WA2' :  .0007751,
+          'FAMINC' :  4.21e-06,
+              'WE' :  .0230147,
+               'K' :   .131184,
+           'const' :  1.406577
+           }
+
+    stata_lambda_coef = -.9595565
+    stata_lambda_stderr = 1.269555
+
+    stata_rho = -0.30079
+    stata_sigma = 3.1901153
+
+    # predictions on first and last rows of data
+    y_new_firstobs = 4.269959
+
+    y_new_lastobs = 3.441829
+
+
+    retdict = {
+        'stata_reg_coef': stata_reg_coef,
+        'stata_reg_stderr': stata_reg_stderr,
+        'stata_select_coef': stata_select_coef,
+        'stata_select_stderr': stata_select_stderr,
+        'stata_lambda_coef': stata_lambda_coef,
+        'stata_lambda_stderr': stata_lambda_stderr,
+        'stata_rho': stata_rho,
+        'stata_sigma': stata_sigma,
+        'y_new_firstobs': y_new_firstobs,
+        'y_new_lastobs': y_new_lastobs
+        }
+
+    return retdict
+
+
+def _check_heckman_to_stata(
+        stata_reg_coef_arr, stata_reg_stderr_arr,
+        stata_select_coef_arr, stata_select_stderr_arr, 
+        stata_lambda_coef, stata_lambda_stderr,
+        stata_rho, stata_sigma,
+        heckman_res,
+        TOL=1e-3):
+    # This function checks output from the Heckman module against known Stata output;
+    # an error occurs if they are different, otherwise function will simply return.
+    # This function will not perform assertion test on a particular estimate if None 
+    # is inputted.
+
+    test_list = [
+        {'trueval' : stata_reg_coef_arr, 'estval': heckman_res.params},
+        {'trueval' : stata_reg_stderr_arr, 
+            'estval': np.sqrt(np.diag(heckman_res.scale*heckman_res.normalized_cov_params))},
+        {'trueval' : stata_select_coef_arr, 'estval': heckman_res.select_res.params},
+        {'trueval' : stata_select_stderr_arr, 
+            'estval': np.sqrt(np.diag(heckman_res.select_res.scale*heckman_res.select_res.normalized_cov_params))},
+        {'trueval' : stata_lambda_coef, 'estval': heckman_res.param_inverse_mills},
+        {'trueval' : stata_lambda_stderr, 'estval': heckman_res.stderr_inverse_mills},
+        {'trueval' : stata_rho, 'estval': heckman_res.corr_eqnerrors},
+        {'trueval' : stata_sigma, 'estval': heckman_res.var_reg_error}
+        ]
+
+    for test_set in test_list:
+        t = np.array(test_set['trueval'])
+        e = np.array(test_set['estval'])
+
+        if t is not None and t.ndim>0:
+            assert_( all(np.atleast_1d( (t-e)/t < TOL )) )
```

Comment:
use `assert_allclose` from numpy.testing, which makes it simpler, it has rtol and atol